src/bin/cargo/cli.rs

```diff
@@ -27,6 +27,14 @@ lazy_static::lazy_static! {
 pub fn main(config: &mut LazyConfig) -> CliResult {
     let args = cli().try_get_matches()?;
 
+    // Update the process-level notion of cwd
+    // This must be completed before config is initialized
+    assert_eq!(config.is_init(), false);
+    if let Some(new_cwd) = args.get_one::<std::path::PathBuf>("directory") {
+        std::env::set_current_dir(&new_cwd)
+            .with_context(|| "could not change to requested directory")?;
```

Comment:
Either change to `.context("…")` for literal string or make the message show the directory path.